examples/C++/DDS/BasicConfigurationExample/BasicConfigurationSubscriber.cpp

```diff
@@ -0,0 +1,250 @@
+#include <fastdds/dds/subscriber/SampleInfo.hpp>
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/rtps/transport/shared_mem/SharedMemTransportDescriptor.h>
+#include <fastdds/rtps/transport/UDPv4TransportDescriptor.h>
+#include <fastrtps/attributes/ParticipantAttributes.h>
+#include <fastrtps/attributes/SubscriberAttributes.h>
+
+#include "BasicConfigurationSubscriber.h"
+
+using namespace eprosima::fastdds::dds;
+using namespace eprosima::fastdds::rtps;
+
+std::atomic<bool> HelloWorldSubscriber::stop_(false);
+std::mutex HelloWorldSubscriber::terminate_cv_mtx_;
+std::condition_variable HelloWorldSubscriber::terminate_cv_;
+
+HelloWorldSubscriber::HelloWorldSubscriber()
+    : participant_(nullptr)
+    , subscriber_(nullptr)
+    , topic_(nullptr)
+    , reader_(nullptr)
+    , type_(new HelloWorldPubSubType())
+{
+}
+
+bool HelloWorldSubscriber::is_stopped()
+{
+    return stop_;
+}
+
+void HelloWorldSubscriber::stop()
+{
+    stop_ = true;
+    terminate_cv_.notify_all();
+}
+
+bool HelloWorldSubscriber::init(
+        const std::string& topic_name,
+        uint32_t max_messages,
+        uint32_t domain,
+        const std::string& transport,
+        bool reliable,
+        bool transient)
+{
+    DomainParticipantQos pqos;
+    pqos.name("Participant_sub");
+
+    // TRANSPORT CONFIG
+    if (!transport.empty())
+    {
+        pqos.transport().use_builtin_transports = false;
+        // SHARED-MEMORY
+        if (transport == "shm")
+        {
+            auto shm_transport = std::make_shared<SharedMemTransportDescriptor>();
+            pqos.transport().user_transports.push_back(shm_transport);
+        }
+        // UDP
+        else
+        {
+            auto udp_transport = std::make_shared<UDPv4TransportDescriptor>();
+            pqos.transport().user_transports.push_back(udp_transport);
+        }
+    }
+
+    // CREATE THE PARTICIPANT
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(domain, pqos);
+
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    // REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    // CREATE THE SUBSCRIBER
+    subscriber_ = participant_->create_subscriber(SUBSCRIBER_QOS_DEFAULT, nullptr);
+
+    if (subscriber_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE TOPIC
+    topic_ = participant_->create_topic(
+        topic_name,
+        "HelloWorld",
+        TOPIC_QOS_DEFAULT);
+
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE READER
+    if (max_messages > 0)
+    {
+        listener_.set_max_messages(max_messages);
+    }
+    DataReaderQos rqos = DATAREADER_QOS_DEFAULT;
+    if (!transport.empty())
+    {
+        rqos.data_sharing().off();
+    }
+    else
+    {
+        rqos.data_sharing().automatic();    // default
+    }
+
+    if (reliable)
+    {
+        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
```

Comment:
Add a comment about why is this needed
```suggestion
        rqos.reliability().kind = RELIABLE_RELIABILITY_QOS;
        rqos.history().kind = KEEP_ALL_HISTORY_QOS;
```

Reply:
:+1: